geoapps/contours/params.py

```diff
@@ -0,0 +1,261 @@
+from geoapps.driver_base.params import BaseParams
+
+
+class ContoursParams(BaseParams):
+    """
+    Parameter class for contours application.
+    """
+
+    def __init__(self, input_file=None, **kwargs):
+        self._default_ui_json = deepcopy(default_ui_json)
+        self._defaults = deepcopy(defaults)
+        self._validations = validations
+        self._objects = None
+        self._data = None
+        self._interval_min = None
+        self._interval_max = None
+        self._interval_spacing = None
+        self._fixed_contours = None
+        self._window_azimuth = None
+        self._window_center_x = None
+        self._window_center_y = None
+        self._window_width = None
+        self._window_height = None
+        self._resolution = None
+        self._export_as = None
+        self._z_value = None
+        self._ga_group_name = None
+
+        if input_file is None:
+            ui_json = deepcopy(self._default_ui_json)
+            input_file = InputFile(
+                ui_json=ui_json,
+                validations=self.validations,
+                validation_options={"disabled": True},
+            )
+
+        super().__init__(input_file=input_file, **kwargs)
+
+    def set_window_params(self):
+        # Fetch vertices in the project
+        lim_x = [1e8, -1e8]
+        lim_y = [1e8, -1e8]
+
+        obj = self.objects
+        if isinstance(obj, Grid2D):
+            lim_x[0], lim_x[1] = obj.centroids[:, 0].min(), obj.centroids[:, 0].max()
+            lim_y[0], lim_y[1] = obj.centroids[:, 1].min(), obj.centroids[:, 1].max()
+        elif isinstance(obj, (Points, Curve, Surface)):
+            lim_x[0], lim_x[1] = obj.vertices[:, 0].min(), obj.vertices[:, 0].max()
+            lim_y[0], lim_y[1] = obj.vertices[:, 1].min(), obj.vertices[:, 1].max()
+        else:
+            return
+
+        width = lim_x[1] - lim_x[0]
+        height = lim_y[1] - lim_y[0]
+
+        if self.window_center_x is None:
+            self.window_center_x = np.mean(lim_x)
+        if self.window_center_y is None:
+            self.window_center_y = np.mean(lim_y)
+        if self.window_width is None:
+            self.window_width = width
+        if self.window_height is None:
+            self.window_height = height
+
+    @property
+    def window(self) -> dict[str, float] | None:
+        """Returns window dictionary"""
+        self.set_window_params()
+        win = {
+            "azimuth": self.window_azimuth,
+            "center_x": self.window_center_x,
+            "center_y": self.window_center_y,
+            "width": self.window_width,
+            "height": self.window_height,
+            "center": [self.window_center_x, self.window_center_y],
+            "size": [self.window_width, self.window_height],
+        }
+        check_keys = ["azimuth", "center_x", "center_y", "width", "height"]
+        no_data = any([v is None for k, v in win.items() if k in check_keys])
```

Comment:
Just a quick comment @jamiefb , I had to change the logic here to make it work. Otherwise the `all` would still return values for window (0 by default). Merging...